File: backend/HorusAPI/HorusAPI/Dto/UserDto.cs
using HorusAPI.Entities;

namespace HorusAPI.Dto
{
    /// <summary>
    /// User account dto
    /// </summary>
    public class UserDto
    {
        /// <summary>
        /// The account name
        /// </summary>
        public string? Name { get; set; }

        /// <summary>
        /// the account password
        /// </summary>
        public string? Password { get; set; }

        /// <summary>
        /// the account email
        /// </summary>
        public string? Email { get; set; }

        /// <summary>
        /// the account role
        /// </summary>
        public Role? Role { get; set; }
    }
}

File: frontend/horus/src/app/dtos/userDto.ts
export class User {
    name: string | undefined;
    email: string | undefined;
    role: number | undefined;
    password: string | undefined;

    roleName() : string {
        switch (this.role) {
            case 0:
                return 'Admin';
            case 1:
                return 'User';
            case 2:
                return 'Developer';
            default:
                return '';
        }
    }
}